pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -202,7 +209,7 @@ let
     inherit kernelPatches randstructSeed extraMakeFlags extraMeta configfile;
     pos = builtins.unsafeGetAttrPos "version" args;
 
-    config = { CONFIG_MODULES = "y"; CONFIG_FW_LOADER = "m"; };
+    config = { CONFIG_MODULES = "y"; CONFIG_FW_LOADER = "m"; } // lib.optionalAttrs withRust { CONFIG_RUST = "y"; };
```

Comment:
It seems like you're adding CONFIG_RUST in multiple places here... something's not right.

Reply:
This is unfortunate, but we pass it along for the same reason we pass `CONFIG_MODULES` along. In `manual-config.nix` we would need to look into the config file to figure this out, which we can't, because this would imply IFD.

I don't think there is a beautiful solution here.

Maybe @K900 has an opinion here as well.